manager/src/main/java/io/gingersnapproject/search/SearchBackendFactory.java

```diff
@@ -0,0 +1,30 @@
+package io.gingersnapproject.search;
+
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import io.gingersnapproject.search.opensearch.OpenSearchBackend;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.elasticsearch.client.RestClient;
+
+public class SearchBackendFactory {
+
+    @ConfigProperty(name = SearchBackend.PROPERTY, defaultValue = "null")
```

Comment:
Can this just not use the Instance field of the SearchBackend? I don't like doing our own factory logic here tbh. Also adding a new impl doesn't require any changes other than the new SearchBackend impl.